store/CSDK/test/README.md

```diff
@@ -0,0 +1,25 @@
+# Example of generate c++ code for file-format using thrift compiler
+
+### File List
+* build_carbondata_test.sh - Generate c++ code and compile
+* carbondata_test.cpp - Parsing .carbonindex using generated c++ code
+
+### Dependencies
+* thrfit 0.9.3
+* gcc 5.4
+* pkg-config
+
+### Usage
+1. build the carbondata_test
+
+```shell
+    sh build_carbondata_test.sh

```

Comment:
lots of errors: 

```
1 error generated.
In file included from carbondata_test.cpp:25:
In file included from ./gen-cpp/carbondata_index_types.h:17:
/usr/local/Cellar/thrift/0.9.3/include/thrift/cxxfunctional.h:95:18: error: no member named 'bind' in namespace 'std'; did you mean 'find'?
    using ::std::bind;
          ~~~~~~~^~~~
                 find
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/algorithm:994:1: note: 'find' declared here
find(_InputIterator __first, _InputIterator __last, const _Tp& __value_)
^
1 error generated.
clang: error: no such file or directory: 'carbondata_constants.o'
clang: error: no such file or directory: 'carbondata_index_constants.o'
clang: error: no such file or directory: 'carbondata_index_merge_constants.o'
clang: error: no such file or directory: 'carbondata_index_merge_types.o'
clang: error: no such file or directory: 'carbondata_index_types.o'
clang: error: no such file or directory: 'carbondata_test.o'
clang: error: no such file or directory: 'carbondata_types.o'
clang: error: no such file or directory: 'dictionary_constants.o'
clang: error: no such file or directory: 'dictionary_metadata_constants.o'
clang: error: no such file or directory: 'dictionary_metadata_types.o'
clang: error: no such file or directory: 'dictionary_types.o'
clang: error: no such file or directory: 'schema_constants.o'
clang: error: no such file or directory: 'schema_types.o'
clang: error: no such file or directory: 'sort_index_constants.o'
clang: error: no such file or directory: 'sort_index_types.o'
```

Reply:
It seems that your development environment does not have libstdc++ installed， it is included in gcc.

to run this file need
* thrfit 0.9.3
* gcc 5.4
* pkg-config

Reply:
my laptop is max, and have gcc,

```
localhost:pycarbon xubo$ gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

Reply:
It's ok after add '-std=c++11' in CFLAGS:
export CFLAGS=`pkg-config --cflags thrift`" -g3 -O0 -std=c++11".

Please validate it in your side. Please add it in this PR if your side is ok too.

Reply:
add '-std=c++11' in CFLAGS